subsys/bluetooth/host/id.c

```diff
@@ -36,6 +37,10 @@ struct bt_adv_id_check_data {
 #if defined(CONFIG_BT_OBSERVER) || defined(CONFIG_BT_BROADCASTER)
 const bt_addr_le_t *bt_lookup_id_addr(uint8_t id, const bt_addr_le_t *addr)
 {
+	CHECKIF(addr == NULL) {
```

Comment:
Since you move from asserts to CHECKIF, the commit message should also be updated